sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample09_MockingEventHubPublisher.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Producer;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests.Snippets
+{
+    [TestFixture]
+    public class Sample09_MockingEventHubPublisher
+    {
+        /// <summary>
+        ///     Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        [Test]
+        public async Task EventBatch()
+        {
+            #region Snippet:EventHubs_Sample09_EventBatchMock
+
+            var size = 1024;
+            var store = new List<EventData> { new EventData(Array.Empty<byte>()), new EventData(Array.Empty<byte>()) };
```

Comment:
I wouldn't focus on pre-populating the batch via the store;  its more intended to let developers peek into it and confirm what they've added to the batch.